### PR TITLE
Panic Attack

### LINKED ISSUE #107: Panic Attack
The robot is to go through a panic attack:

- This panic attack is to happen after the "water plants" mission
- Panic attack essentially refers to the robot player realising there is too much work to do by himself
- Consists of camera shaking, red tint, and some noise
- Will also lock game for the duration of the panic attack
- The robot is a soft boy

### PR DESCRIPTION
**Pull request by Aylinx13**
There is a panic attack sequence that is triggered once the robot completes all missions, a flickering red panel along with erratic thoughts are displayed (with accompanying sound). This sequence is run for a few seconds and then a calm down sequence is applied where the other robots contact the player to say reinforcements are on their way. This finishes with the win panel being displayed.

closes #107

### CODE REVIEW

File: Assets/Scenes/Aylin.unity
Comment (Amal-Mathew204) [inline Assets/Scenes/Aylin.unity:1]: Remove Your Scene from this branch

File: Assets/Scripts/Game/PanicTrigger.cs [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+        [SerializeField] private float thoughtDisplayDuration = 0.5f;
+        [SerializeField] private float fadeInDuration = 0.2f;
+        [SerializeField] private float fadeOutDuration = 0.2f;
+        [SerializeField] private float initialPanicDelay = 1f;
+
+        private string[] _thoughts = new string[]
+        {
+            "This is too much work...", "How am i supposed to to this by myself", "I just can't.", "I'm going to fail!",
+            "What do I do?"
+        };
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Game/PanicTrigger.cs:43]: Not all private variables follow _camelCase

File: Assets/Scripts/Game/PanicTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+        #endregion
+
+        #region Initialisation
+        /// <summary>
+        /// Locates and assigns the PlayerInput component from the player GameObject tagged "Player"
+        /// Retrieves the Image component from the assigned panicPanel
+        /// </summary>
+        void Start()
+        {
+            _playerInput = GameObject.FindGameObjectWithTag("Player").GetComponent<PlayerInput>();
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Game/PanicTrigger.cs:53]: As discussed previously this can be removed

File: Assets/Scripts/Game/PanicTrigger.cs [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+                PlayerManager.Instance.DisablePlayerMovement();
+                Cursor.lockState = CursorLockMode.None;
+                Cursor.visible = true;
+                PlayerManager.Instance.SwitchToFirstPerson();
+                gameState.PauseBatteryConsumption(); 
+            }
+            else
+            {
+                // Time.timeScale = 1; // Resume the game
+                PlayerManager.Instance.EnablePlayerMovement();
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Game/PanicTrigger.cs:127]: remove commented out lines of code

File: Assets/Scripts/Game/PanicTrigger.cs [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+                Cursor.lockState = CursorLockMode.Locked;
+                Cursor.visible = false;
+                PlayerManager.Instance.SwitchToThirdPerson();
+                gameState.ResumeBatteryConsumption();
+            }
```
Comment (Amal-Mathew204) [inline Assets/Scripts/Game/PanicTrigger.cs:132]: If the game is going to end through panic attack this is redundant and can lead to the potential of the player still being able to lose

**Review by Amal-Mathew204 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Game/PanicTrigger.cs (hunk 1 above)
Comment (Aylinx13) [reply Assets/Scripts/Game/PanicTrigger.cs:43]: they all doo??

**Review by Amal-Mathew204 — APPROVED** [review]

**Pull request merged**